### PR TITLE
Add sampler solver

### LINKED ISSUE #112: Convexification of network outputs (1D)

### PR DESCRIPTION
**Pull request by SebastianGuadalupe**
Closes #112.

### CODE REVIEW

File: src/nnops.jl [hunk 1]
```diff
@@ -93,3 +93,17 @@ end
...
+    nsamples::Int64 = 10000
+end
+
+function NeuralVerification.forward_network(solver::SampledApprox, nnet, input)
+    samples = sample(input, solver.nsamples)
+    outputs = Vector()
+    for i in samples
+        push!(outputs, NV.compute_output(nnet, i))
+    end
+    return Interval(first.(convex_hull(UnionSetArray(Singleton.(outputs)))))
```
Comment (mforets) [inline src/nnops.jl:108]: i think this is equivalent to `Interval(extrema(outputs)...)`

File: src/nnops.jl [hunk 2]
```diff
@@ -93,3 +93,17 @@ end
...
 end
+
+#### Solver using the CH of the sampled outputs as an inner approx of the real output
+@with_kw struct SampledApprox <: Solver
+    nsamples::Int64 = 10000
+end
+
+function NeuralVerification.forward_network(solver::SampledApprox, nnet, input)
+    samples = sample(input, solver.nsamples)
+    outputs = Vector()
```
Comment (mforets) [inline src/nnops.jl:104]: under the 1D assumption this is just a `Vector{Float64}(undef, solver.nsamples)`.

File: src/utils.jl [hunk 3]
```diff
@@ -266,3 +266,11 @@ function random_control_problems(n::Integer, m::Integer; ivp=nothing, period=0.1
 
     return problems
 end
+
+include("nnops.jl")
```
Comment (mforets) [inline src/utils.jl:270]: files should be included in the main `.jl`

File: src/utils.jl [hunk 4]
```diff
@@ -266,3 +266,11 @@ function random_control_problems(n::Integer, m::Integer; ivp=nothing, period=0.1
...
+# relative size between the set-based output and the (CH of) sampled output
+function relative_size_nnsol(X0, nsamples, solver, controller)
```
Comment (mforets) [inline src/utils.jl:272]: i think we can call this `relative_size` and change the signature to `relative_size(X0, nsampler, controller, solver=Ai2z())`.

Comment (mforets) [reply src/utils.jl:272]: maybe add an assertion `@assert dim(X0) == 1`

File: src/nnops.jl [hunk 5]
```diff
@@ -93,3 +93,17 @@ end
...
     return translate(X, fill(normalization.shift, dim(X)))
 end
+
+#### Solver using the CH of the sampled outputs as an inner approx of the real output
+@with_kw struct SampledApprox <: Solver
+    nsamples::Int64 = 10000
+end
+
+function NeuralVerification.forward_network(solver::SampledApprox, nnet, input)
+    samples = sample(input, solver.nsamples)
```
Comment (mforets) [inline src/nnops.jl:103]: also here let's `@assert` that the output dimension is 1

File: src/nnops.jl [hunk 6]
```diff
@@ -93,3 +93,17 @@ end
 function apply(normalization::UniformAdditiveNormalization, X::LazySet)
     return translate(X, fill(normalization.shift, dim(X)))
 end
+
+#### Solver using the CH of the sampled outputs as an inner approx of the real output
+@with_kw struct SampledApprox <: Solver
+    nsamples::Int64 = 10000
```
Comment (mforets) [inline src/nnops.jl:99]:
```suggestion
    nsamples::Int = 10000
```

File: src/nnops.jl [hunk 7]
```diff
@@ -115,3 +118,20 @@ end
...
 end
+
+# solver using the CH of the sampled outputs as an inner approx of the real output
+@with_kw struct SampledApprox <: Solver
+    nsamples::Int = 10000
+end
+
+function NeuralVerification.forward_network(solver::SampledApprox, nnet, input)
+    samples = sample(input, solver.nsamples)
+    MIN = first(NV.compute_output(nnet, sample(input)))
```
Comment (mforets) [inline src/nnops.jl:129]: since we are using `Float64`, you could also initialize with `MIN = Inf` and `MAX = -Inf`

**Review by mforets — APPROVED** [review]

**Review by schillic — APPROVED** [review]

**Pull request merged**